Assignment 2/src/Vault.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "yield-utils-v2/token/IERC20.sol";
+
+/// @title A vault that can store tokens.
+/// @notice This vault allows users to deposit and withdraw tokens.
+/// @author Thomas den Hollander
+/// @dev Deposits are implemented using a mapping. This is the only (mutable)
+///     state that is kept by the contract.
+///
+///     Amounts are stored with the same decimals as the token contract that
+///     was used to initialize the `Vault`.
+contract Vault {
+    /// @notice The token that is stored in the vault.
+    IERC20 immutable public token;
+
+    /// @notice This event is emitted when a successful deposit is made. It
+    ///     contains the account that made the deposit and the amount that was
+    ///     deposited.
+    event Deposit(address indexed account, uint256 deposited);
+
+    /// @notice This event is emitted when a successful withdrawal is
+    ///     performed. The first argument is the account that made the
+    ///     withdrawal and the second argument is the amount that was
+    ///     withdrawn.
+    event Withdrawal(address indexed account, uint256 withdrawn);
+
+    /// @notice Deposits made using `deposit` can be reverted with this error
+    ///     if the transfer failed.
+    /// @dev This error is thrown if `transferFrom` returns `false`. Instead of
+    ///     returning false however, it is possible that the `IERC20` reverts,
+    ///     for example when attempting to transfer more than the allowance.
+    ///     So: not all failed deposits will throw this error.
+    error DepositFailed(address account, uint256 amount);
```

Comment:
I'm not used to seeing errors here so I got confused and thought these were events at first.  I would suggest either moving them outside the contract to the top level (on larger ctrcts w a lot of errors, they can even have their own file) or else using section headers for each section within the contract